### PR TITLE
#5053: add scroll into view brick

### LINKED ISSUE #5053: PLACEHOLDER: Add scrollIntoView brick

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Closes #5053 
- Matches the API of Element.scrollIntoView: https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView

## Demo

- https://www.loom.com/share/a73715b055e94368bc1583cb5fc48a21

## Checklist

- [x] Add tests
- [x] Designate a primary reviewer: @fregante

### CODE REVIEW

File: src/blocks/effects/scrollIntoView.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      isRootAware: true,
+      blockId: this.id,
+    });
+
+    if ($elements.length > 1) {
+      throw new MultipleElementsFoundError(selector);
+    }
+
+    if ($elements.length === 0) {
+      throw new NoElementsFoundError(selector);
```
Comment (fregante) [inline src/blocks/effects/scrollIntoView.ts:97]: I think this logic exists elsewhere, like "requireOneElement" I think

File: src/blocks/effects/scrollIntoView.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    );
+  }
+
+  inputSchema: Schema = propertiesToSchema(
+    {
+      selector: {
+        type: "string",
+        format: "selector",
+        description:
+          "A jQuery selector, or leave blank to scroll to the target element",
```
Comment (fregante) [inline src/blocks/effects/scrollIntoView.ts:43]: What's a target element? How does it flow into this function? I only see it "selector" as the input

**Review by fregante — APPROVED** [review]

File: src/blocks/effects/scrollIntoView.ts (hunk 2 above)
Comment (twschiller) [reply src/blocks/effects/scrollIntoView.ts:43]: The brick is root aware. As of 1.7.16, you can pass an element reference as the target for a brick to run against: https://github.com/pixiebrix/pixiebrix-extension/pull/4875

As part of that, we made the "selector" field optional on nearly all the bricks

image

File: src/blocks/effects/scrollIntoView.ts (hunk 1 above)
Comment (twschiller) [reply src/blocks/effects/scrollIntoView.ts:97]: There's a requireSingleElement function, but it doesn't work quite right with the root mode and optional selector logic. We'll likely need to introduce another function

Comment (twschiller) [reply src/blocks/effects/scrollIntoView.ts]: I just pushed with a refactored method

**Pull request merged**